kusto/ingest/internal/filesystem/filesystem.go

```diff
@@ -91,6 +105,53 @@ func (i *Ingestion) Local(ctx context.Context, from string, props properties.All
 	return nil
 }
 
+// Reader uploads a file via an io.Reader.
+func (i *Ingestion) Reader(ctx context.Context, reader io.Reader, props properties.All) error {
+	to, err := i.upstreamContainer()
+	if err != nil {
+		return err
+	}
+
+	resources, err := i.mgr.Resources()
+	if err != nil {
+		return err
+	}
+
+	// We want to check the queue size here so so we don't upload a file and then find we don't have a Kusto queue to stick
+	// it in. If we don't have a container, that is handled by containerQueue().
+	if len(resources.Queues) == 0 {
+		return errors.ES(errors.OpFileIngest, errors.KBlobstore, "no Kusto queue resources are defined, there is no queue to upload to").SetNoRetry()
+	}
+
+	blobName := fmt.Sprintf("%s_%s_%s_%s.gz", filepath.Base(uuid.New().String()), nower(), i.db, i.table)
+
+	// Here's how to upload a blob.
+	blobURL := to.NewBlockBlobURL(blobName)
+
+	gstream := gzip.New()
```

Comment:
shouldn't we make sure we are not double-zipping?

Reply:
I don't think so.  That is kind of an infinite loop.  They can do other compression other than gzip.  

We tell them in the docs specifically not to compress the stream (and to be fair, most people wouldn't do that anyways).